src/System.CommandLine.Rendering/SystemTerminal.cs

```diff
@@ -87,25 +58,11 @@ public void TryEnableVirtualTerminal()
             _virtualTerminalMode = VirtualTerminalMode.TryEnable();
         }
 
-        private void ResetConsole()
+        protected override void Dispose(bool disposing)
```

Comment:
Shouldn't we only dispose `_virtualTerminalMode` if `disposing` is true?

Reply:
Whether this is called via an explicit `Dispose` or finalizer, we want to make sure that the virtual terminal mode gets reset.

Reply:
I don't believe that is correct. There is not defined order that the GC will clean up managed objects. So if this is invoked from a finalizer the `_virtualTerminalMode` may have already been GC'd. In fact, looking at the code closer, this is probably an indication that we should implement the Disposable pattern on `VirtualTerminalMode` as well.

Reply:
Good point. I'll move the finalizer to `VirtualTerminalMode`.